README.md

```diff
@@ -1,2 +1,406 @@
+
+Every `@Instantiable`-decorated type must be:
+
+1. `public` or `open`
+
+2. Have a `public init(…)` or `open init(…)` that receives every injectable property
+
+The `@Instantiable` macro guides engineers through satisfying these requirements with build-time FixIts.
+
+Here is a sample `UserService` implementation that is `@Instantiable`:
+
+```swift
+import SafeDI
+
+public final class UserService {
+
+    // MARK: Initialization
+
+    /// Public, memberwise initializer that takes each injected property.
+    public init(authService: AuthService, securePersistentStorage: SecurePersistentStorage) {
+        self.authService = authService
+        self.securePersistentStorage = securePersistentStorage
+    }
+
+    // MARK: UserService
+
+    public private(set) lazy var user: User? = loadPersistedUser() {
+        didSet {
+            persistUserData()
+        }
+    }
+
+    public func login(username: String, password: String) async throws -> User {
+        let user = try await authService.login(username: username, password: password)
+        self.user = user
+        return user
+    }
+
+    // MARK: Private
+
+    /// An auth service instance that is instantiated when the `UserService` is instantiated.
+    @Instantiated
+    private let authService: AuthService
+
+    /// An instance of secure, persistent storage that is instantiated further up the SafeDI dependency tree.
+    @Received
+    private let securePersistentStorage: SecurePersistentStorage
+
+    private func loadPersistedUser() -> User? {
+        securePersistentStorage["user", ofType: User.self]
+    }
+
+    private func persistUserData() {
+        securePersistentStorage["user"] = user
+    }
+}
+```
+
+#### Making protocols `@Instantiable`
+
+While it is not necessary to utilize protocols with SafeDI, protocol-driven development aids both testability and dependency inversion. The `@Instantiable` macro has a parameter `fulfillingAdditionalTypes` that enables any concrete `@Instantiable` type to fulfill properties that are declared as conforming to a protocol (or superclass) type. Here’s a sample implementation of a protocol-backed, `@Instantiable` `UserService`:
+
+```swift
+import SafeDI
+
+/// A protocol that defines a UserService.
+public protocol UserService {
+    var user: User? { get }
+    func login(username: String, password: String) async throws -> User
+}
+
+/// A default implementation of `UserService` that can fulfill `@Instantiated`
+/// properties of type `UserService` or `DefaultUserService`.
+@Instantiable(fulfillingAdditionalTypes: [UserService.self])
```

Comment:
This is fufilling **additional** types because `DefaultUserService` will be used to fulfill dependencies of type `UserService` _and_ `DefaultUserService`. Is that right? Asking to make sure I understand clearly.

Reply:
Correct!